src/main/java/space/npstr/magma/events/api/WebSocketClosedEvent.java

```diff
@@ -0,0 +1,49 @@
+package space.npstr.magma.events.api;
+
+import space.npstr.magma.Member;
+import space.npstr.magma.events.audio.ws.CloseCode;
+
+import javax.annotation.Nullable;
+
+/**
+ * This event is fired when an audio web socket is closed. However, this event is not fired if we are trying to
+ * resume (i.e reconnect) automatically unless resuming causes a new socket to close.
+ */
+@SuppressWarnings("unused")
+public class WebSocketClosedEvent implements MagmaEvent {
```

Comment:
Please implement this class via immutables.org, like all the other value objects/events in this project.